moveos/moveos-object-runtime/src/runtime_object.rs

```diff
@@ -641,6 +642,52 @@ impl RuntimeObject {
 
         Ok((fields, Some(total_bytes_len)))
     }
+
+    /// List fields of the object from the state store.
+    pub fn list_fields(
+        &self,
+        layout_loader: &dyn TypeLayoutLoader,
+        resolver: &dyn StatelessResolver,
+        cursor: Option<FieldKey>,
+        limit: usize,
+        field_type: &Type,
+    ) -> PartialVMResult<(Vec<Value>, Option<Option<NumBytes>>)> {
+        debug!(
+            "enter list_fields, cursor:{:?}, limit:{:?}, field_type:{:?}",
+            cursor, limit, field_type
+        );
+        let expect_value_type = layout_loader.type_to_type_tag(field_type)?;
+        debug!("expect_value_type: {:#?}", expect_value_type);
+        let fields_with_objects =
+            self.list_field_objects_from_db(layout_loader, resolver, cursor, limit)?;
+        debug!(
+            "list_field_objects_from_db length: {}",
+            fields_with_objects.len()
+        );
+        let mut fields = Vec::with_capacity(fields_with_objects.len());
+        let mut total_bytes_len = NumBytes::zero();
+
+        for (key, db_obj, bytes_len_opt) in fields_with_objects {
+            let (value, bytes_len) = if let Some(cached_field) = self.get_loaded_field(&key) {
+                (
+                    cached_field.borrow_value(Some(&expect_value_type))?,
```

Comment:
We use borrow there, the return value should be `&V`? But Move does not support `vector<&V>` yet. Maybe we should support `list_field_names` in Move, and the Move code to `borrow_field` via the name.